### ISSUE TITLE
Add an option to announce extra routes

### ISSUE THREAD

**Issue by Gyosa3** [lmagyar/homeassistant-addon-tailscale-beta#5]
Hi,

I'm wondering if you would like to add an extra configuration option where one can define the routes to announce. I am not fancy of the "auto-discovery" function built in the addon and I'd like to override it when needed.

Something in the area of:

- when no option is set, autodiscover the local LAN where HA is running
- when the option is set to "disabled" (whatever way), do not announce any route
- when option is set, do not autodiscover and instead announce the routes that are set in the option

If you prefer I can open an issue in the official addon repo.

**Comment by Gyosa3** [comment]
This is how it looks like in Tsujamin's add-on (https://github.com/tsujamin/hass-addons) that I use on the distant node:
![image](https://github.com/lmagyar/homeassistant-addon-tailscale-beta/assets/51777889/8b155c26-a449-446a-b5a4-4048cc63a8e0)

I'm looking at the code of your addon and although I have found the auto-discovery function in `usr/bin/subnet-routes` I fail to see where you add the advertised routes to the executable.

Something like (From Tsujamin's code)
```
if bashio::config.has_value 'advertise_routes'; then
    TAILSCALE_FLAGS+=('-advertise-routes' "$(bashio::config 'advertise_routes')")
fi
```

that could be added for the 3rd bullet above

**Comment by lmagyar** [comment]
Yes, this is the exact same situation/solution that happened to samba add-on, to run not only on the HA Supervisor managed interfaces.

HA team insists on ***not*** dealing with networks that are not managed by HA Supervisor. Only interfaces listed by `ha network info` can be used (eg. eth0, wlan0).

It didn't get merged, I maintain a samba add-on fork basically only for this extra option, but at least I use this extra option.

I plan to finish maintaining this tailscale fork after the official add-on is released, so I prefer not to add config options that has 0% chance to be merged.

Though I leave this issue open now. After you manage to make your networking setup working without `advertise_routes` and `snat_subnet_routes` options available, and only these are what required, I maybe add them.

**Comment by Gyosa3** [comment]
Ok, I see the point here, can't blame the HA team. I'll rethink my setup see how bad I need this.

Thanks!
